test/core/runtime/wasm_test.cpp

```diff
@@ -0,0 +1,170 @@
+      std::shared_ptr<const kagome::runtime::CoreApiFactory>,
+      std::shared_ptr<const kagome::runtime::MemoryProvider>,
+      std::shared_ptr<kagome::runtime::TrieStorageProvider>) const override {
+    return std::make_unique<kagome::host_api::HostApiMock>();
+  }
+};
+auto host_api_factory = std::make_shared<StubHostApiFactory>();
+
+namespace kagome::runtime::wasm_edge {
+  auto make(bool compile) {
+    using M = ModuleFactoryImpl;
+    using E = M::ExecType;
+    return std::make_shared<M>(hasher,
+                               host_api_factory,
+                               trie_storage,
+                               nullptr,
+                               nullptr,
+                               compile ? E::Compiled : E::Interpreted);
+  }
+  auto interpreter = [] { return make(false); };
+  auto compiler = [] { return make(true); };
+}  // namespace kagome::runtime::wasm_edge
+
+std::unique_ptr<wabt::Module> wat_to_module(std::span<const uint8_t> wat) {
+  wabt::Result result;
+  wabt::Errors errors;
+  std::unique_ptr<wabt::WastLexer> lexer =
+      wabt::WastLexer::CreateBufferLexer("", wat.data(), wat.size(), &errors);
+  if (Failed(result)) {
+    throw std::runtime_error{"Failed to parse WAT"};
+  }
+
+  std::unique_ptr<wabt::Module> module;
+  wabt::WastParseOptions parse_wast_options{{}};
+  result =
+      wabt::ParseWatModule(lexer.get(), &module, &errors, &parse_wast_options);
+  if (Failed(result)) {
+    throw std::runtime_error{"Failed to parse module"};
+  }
+  return module;
+}
+
+std::vector<uint8_t> wat_to_wasm(std::span<const uint8_t> wat) {
+  auto module = wat_to_module(wat);
+  wabt::MemoryStream stream;
+  if (wabt::Failed(wabt::WriteBinaryModule(
+          &stream,
+          module.get(),
+          wabt::WriteBinaryOptions{{}, true, false, true}))) {
+    throw std::runtime_error{"Failed to write binary wasm"};
+  }
+  return std::move(stream.output_buffer().data);
+}
+
+auto fromWat(std::string_view wat) {
+  return wat_to_wasm(kagome::str2byte(wat));
+}
+
+void test(std::string name,
+          const kagome::runtime::ModuleFactory &factory,
+          bool is_interpreter = false) {
+  using Instance = std::shared_ptr<kagome::runtime::ModuleInstance>;
+  auto read = [&](Instance instance) {
+    return fmt::format("{:x}",
+                       instance->getEnvironment()
+                           .memory_provider.get()
+                           ->getCurrentMemory()
+                           .value()
+                           .get()
+                           .view(0, 4)
+                           .value());
+  };
+  auto compile = [&](bool bulk) -> Instance {
+    auto path = fmt::format("test-wasm-{}-{}", name, bulk);
+    auto context_params = kagome::runtime::RuntimeContext::ContextParams();
+    context_params.wasm_ext_bulk_memory = bulk;
+    if (not is_interpreter) {
+      auto wat_path = kagome::filesystem::path(__FILE__).parent_path().string()
```

Comment:
`__FILE__` doesn't work, use cmake
```cmake
target_compile_definitions(wasm_test PRIVATE PROJECT_SOURCE_DIR="${PROJECT_SOURCE_DIR}")
```

Reply:
What do you mean by "doent't work" ? It works at least locally, also __FILE__ is used in bunch of other tests like here https://github.com/qdrvm/kagome/blob/56f816c0ff54767356df5ef785b5ef9a9aca82ff/test/core/application/chain_spec_test.cpp#L44

Reply:
What to you mean by "doesn't work"? It works locally, also __FILE__ in used in other tests like here https://github.com/qdrvm/kagome/blob/56f816c0ff54767356df5ef785b5ef9a9aca82ff/test/core/application/chain_spec_test.cpp#L44

Reply:
`__FILE__` is not always absolute
```bash
echo "__FILE__" > foo.h
cc -E foo.h
```
prints `"foo.h"`

Reply:
Used filesystem::absolute